webrtc/RTCPeerConnection-set-description.html

```diff
@@ -0,0 +1,395 @@
+    if(matches === null) {
+      return 0;
+    } else {
+      return matches.length;
+    }
+  }
+
+  function similarMediaDescriptions(sdp1, sdp2) {
+    if(sdp1 === sdp2) {
+      return true;
+    } else if(countAudioLine(sdp1) !== countAudioLine(sdp2)) {
+      return false;
+    } else if(countVideoLine(sdp1) !== countVideoLine(sdp2)) {
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  function assert_defined(object) {
+    if(object === undefined || object === null)
+      assert_unreached('object must not be undefined or null');
+  }
+
+  function isSimilarSdp(sdp1, sdp2) {
+    assert_defined(sdp1);
+    assert_defined(sdp2);
+
+    if(sdp1.type !== sdp2.type) {
+      return false;
+    } else {
+      // We can't do string comparison to the SDP content,
+      // because RTCPeerConnection may return SDP that is
+      // slightly modified from what is given to it due to
+      // ice candidate events or serialization.
+      // To hack around this, if the SDP strings are not
+      // equal, we simply count the media description lines
+      // and if they are the same, we assume it is the same.
+      return similarMediaDescriptions(sdp1.sdp, sdp2.sdp);
+    }
+  }
+
+  function createOfferAndSetLocal(pc, options) {
+    return pc.createOffer(options)
+    .then(offer => {
+      return pc.setLocalDescription(offer)
+      .then(() => offer)
+    });
+  }
+
+  function createOfferAndSetRemote(pc1, pc2, options) {
+    return createOfferAndSetLocal(pc1, options)
+    .then(offer =>
+      pc2.setRemoteDescription(offer)
+      .then(() => offer));
+  }
+
+  function createOfferAndGetAnswer(pc1, pc2, options) {
+    return createOfferAndSetRemote(pc1, pc2, options)
+    .then(offer =>
+      pc2.createAnswer(offer)
+      .then(answer => [offer, answer]))
+  }
+
+  function createAnswerAndSetLocal(pc1, pc2, options) {
+    return createOfferAndGetAnswer(pc1, pc2, options)
+    .then(([offer, answer]) =>
+      pc2.setLocalDescription(answer)
+      .then(() => [offer, answer]));
+  }
+
+  function createAnswerAndSetRemote(pc1, pc2, options) {
+    return createAnswerAndSetLocal(pc1, pc2, options)
+    .then(([offer, answer]) =>
+      pc1.setRemoteDescription(answer)
+      .then(() => [offer, answer]));
+  }
+
+  test(t => {
+    const pc = new RTCPeerConnection;
+    assert_equals(pc.signalingState, 'stable');
+    assert_equals(pc.localDescription, null);
+    assert_equals(pc.currentLocalDescription, null);
+    assert_equals(pc.pendingLocalDescription, null);
+    assert_equals(pc.remoteDescription, null);
+    assert_equals(pc.currentRemoteDescription, null);
+    assert_equals(pc.pendingRemoteDescription, null);
+  }, 'Initial peer connection state');
+
+  promise_test(t => {
+    const pc = new RTCPeerConnection;
+    return createOfferAndSetLocal(pc)
+    .then(offer => {
+      assert_equals(pc.signalingState, 'have-local-offer');
+      assert_true(isSimilarSdp(pc.localDescription, offer));
```

Comment:
Ok. I have added `assert_session_description_equals` and `assert_session_description_not_equals` functions for that.